news: Neovim News #12 - The Work Continues (#248)
neovim/neovim.github.io: _posts/2021-09-26-newsletter.md

### Lua config

Nvim supports `init.lua` as user config since 0.5, and in 0.6. we are working on `nvim -l` to support invoking Lua "main" scripts with `nvim`.

## Language Server Protocol (LSP)

The [Language Server Protocol](https://microsoft.github.io/language-server-protocol/) (LSP) is an
open, JSON-RPC protocol for communication between code editors and
language servers, which provide programming language-specific features such as
completion, hover/tooltips, go to definition, show/go to references, show
method signatures, rename, refactoring.

Nvim's builtin [`vim.lsp`](https://neovim.io/doc/user/lsp.html) [LSP framework](https://www.youtube.com/watch?v=ArwDgvYEZYk)
plus [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) quickstart collection has been a major success,
demonstrating the viability of shipping ambitious ideas in core Nvim using Lua.
Nvim core developers use it to accelerate their work in Nvim's C and Lua source code.

Work on [semantic highlighting](https://github.com/neovim/neovim/pull/14122) continues.

## Tree-sitter

Work on [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) parser integration continues: performance, bug-fixes.

> Also, Neovim itself only provides a (Lua) API for generating and querying the syntax tree using the bundled tree-sitter library; see [`:h treesitter`](https://neovim.io/doc/user/treesitter.html). User-facing features like the ones mentioned above are implemented in plugins like

Reviewer: (note: unfinished sentence here `implemented in plugins like .....`)
maybe use `XXX` to indicate 'fill-me' areas? :eyes: 
```suggestion
Also, Neovim itself only provides a (Lua) API for generating and querying the syntax tree using the bundled tree-sitter library; see [`:h treesitter`](https://neovim.io/doc/user/treesitter.html). User-facing features like the ones mentioned above are implemented in plugins like XXX and XXX.
```

---
layout: newsletter
title: "Neovim News #12 - The Work Continues"
category: newsletter
permalink: /news/2021/09
---

> The Citadel stripped me of it. They found some of my... experiments... too bold.
>
> —Qyburn

Nvim 0.5.1 has been released, it includes many fixes and some breaking changes:

- ...
- ...

## Velocity

Lua-first development is now a central practice in the Neovim project.

We started with everything tucked into `vim.lsp`. The community began heavily
building on `vim.lsp.diagnostic`, it became clear that this "diagnostic"
concept had more general value beyond only LSP.
The development of `vim.diagnostic` diagnostic framework from concept, to
interface design, to delivery was a matter of weeks, thanks to @gpanders.

The `vim.diagnostic` API is available in Nvim master (unreleased 0.6) and we are considering it for 0.5.2 stable release.

> Thoug the Nvim RPC API is bound by a sacred

Reviewer 2: ```suggestion
Though the Nvim RPC API is bound by a sacred
```

> [contract](https://neovim.io/doc/user/api.html#api-contract), in other ares of Nvim we are now more

Reviewer 2: ```suggestion
[contract](https://neovim.io/doc/user/api.html#api-contract), in other areas of Nvim we are now more
```

**Windows 8 or older is unsupported,** particularly for TUI/TTY bug reports.

### Vim patches

Of the over XXX commits in this release, about XXX were patches and runtime updates ported from Vim.

Nvim `master` branch is up-to-date with Vim runtime files.

## Community

In keeping with the motto of this newsletter, one of the most noticeable positive changes was the growth of the community and of new ways of interacting with it.

### Discourse

Previously, support requests and discussions were spread across Reddit, Gitter, and GitHub Discussions and were either ephemeral or hard to search for. We have now consolidated around a new [Neovim Discourse](https://neovim.discourse.group), which is a free and open source forum platform with mailing list and RSS features, in addition to a nice web interface. The Neovim Discourse is an official core project and moderated by core team members.

### Matrix

The official chatroom for Neovim is on [matrix](https://app.element.io/#/room/#neovim:matrix.org).

### Vimconf.live

Community members and Neovim core developers will speak at [Vimconf.live](https://www.vimconf.live).

> ### Twitch

Reviewer 3: ```suggestion
### Streaming
```

> Neovim core developers @vigoux and [@tjdevries](https://www.twitch.tv/teej_dv) continue to stream on [Twitch](https://twitch.tv).

Reviewer 3: ```suggestion
Neovim core developers [@vigoux](https://odysee.com/@vigoux:8) and [@tjdevries](https://www.twitch.tv/teej_dv) continue to stream.
```
